README.MD

```diff
@@ -1,224 +1,277 @@
+
+### What happens when a worker pod goes back up?
+It reuses the data existing before it went down. Then it tries to join the cluster and if the data is fresh enough it will catch up by applying recent changes occurred in the cluster, otherwise it will do full sync like if it was a completely new worker. If it can't find the cluster at all it will try to join the oldest node or node 0 (see below about replication modes).
 
-| SQL | HTTP | 
-|---|---|
-|`kubectl --namespace $NAMESPACE port-forward $WORKER_NAME 7306:9306` | `kubectl --namespace $NAMESPACE port-forward $WORKER_NAME 7308:9308` | 
-
-### Connect to the worker
-
-| SQL | HTTP | 
-|---|---|
-|`mysql -h0 -P7306`| No connection required since HTTP is a stateless protocol |
+### How Do I Delete Data?
+You can truncate or delete tables, manually clean volumes, or delete the entire namespace.
 
-### Create your first table:
+### What Happens When a New Worker Pod is Added?
+- The new pod starts and runs a side process to check for active nodes.
+- It joins the oldest node or the cluster's primary worker.
+- The pod waits until it fully syncs before becoming operational.
 
-| SQL | HTTP | 
-|---|---|
-|`CREATE TABLE idx(title text);`|`curl -X "POST" localhost:7308/sql -d "mode=raw&query=create table idx(title text)"`|
+### What happens if I uninstall the helm chart?
+The volumes will still retain the data and if you install the chart again to the same namespace you'll be able to access it.
 
-### Add table to cluster:
+### What Is the Balancer and How Does It Work?
 
-| SQL | HTTP | 
-|---|---|
-|`ALTER CLUSTER manticore_cluster ADD idx;`|`curl -X "POST" localhost:7308/sql -d "mode=raw&query=alter cluster manticore_cluster add idx"`|
+The balancer in ManticoreSearch is a key component for managing high availability (HA) in a replication cluster. It ensures that read operations are distributed efficiently across available worker pods, while write operations bypass the balancer and are sent directly to the worker pods.
 
-### Add some data to your table:
+#### Key Functions of the Balancer:
+- **Read Operations:** The balancer intelligently routes read queries (`SELECT`) to the most appropriate workers based on their status and availability. This ensures load is distributed evenly across active workers, enhancing cluster efficiency and reliability.
+- **Write Operations:** Writes (`INSERT`, `UPDATE`, etc.) should be manually sent directly to the worker pods, bypassing the balancer.
 
-**SQL**
-```sql
-INSERT INTO manticore_cluster:idx (title) VALUES ('dog is brown'), ('mouse is small');
-```
+#### How the Balancer Works:
+1. **Dynamic Synchronization:**
+   - If `balancer.enabled` is set to `true` (default), the balancer checks the table configuration on the first active worker every 5 seconds (configurable via the `balancer.runInterval` parameter).
+   - If the configuration differs from its current state, the balancer reloads itself to align with the updated tables.
```

Comment:
What exactly is aligned? It's not quite clear.

Reply:
It queries the k8s API to understand what workers are available. This is relevant if one of the workers has fallen off due to a rediness probe.

Reply:
Let's improve this, so it's clear what "align" means.